src/lib/ApiEndpoint.test.ts

```diff
@@ -57,104 +53,213 @@ describe('ApiEndpoint', () => {
-          name: 'foo',
-          alt: 'foo',
-        } as const
-
-        const mockHandleRequest = jest.fn(
-          (): Promise<Result<any, never>> => {
-            return new Promise((resolve) => {
-              resolve(
-                Result.accept({
-                  data: [
-                    {
-                      type: data.type,
-                      id: data.id,
-                      attributes: {
-                        assetType: data.assetType,
-                        name: data.name,
-                        alt: data.alt,
-                      },
-                    },
-                  ],
-                }),
-              )
-            })
+        const endpoint = api.endpoint('/posts', Post)
+
+        const mockHandleRequest = jest.fn().mockResolvedValue(
+          Result.accept({
+            data: [mockData.Post.p1],
+          }),
+        )
+
+        api.controller.handleRequest = mockHandleRequest
+
+        let items
+        try
+        {
+          items = await endpoint.fetch({}, {fields: {Post: ['title', 'content']}})
+        } catch (errors)
+        {
+          console.log(errors)
+          throw errors
+        }
+
+        expect(items.data).toEqual([
+          {
+            id: 'p1',
+            type: 'Post',
+            title: 'Post 1',
+            content: 'foo',
           },
+        ])
+        expect(mockHandleRequest.mock.calls[0][0].href).toEqual('https://www.example.com/api/posts?fields[Post]=content,title');
+      })
+    })
+
+    describe('getRelationshipEntity', () => {
+      it('should fetch a one-to-one relationship', async () => {
+        const url = 'https://www.example.com/api'
+        const api = new Api(new URL(url))
+
+        const endpoint = api.endpoint('/posts', Post)
+        api.register(Author)
+
+        const mockHandleRequest = jest.fn().mockResolvedValue(
+          Result.accept({
+            data: mockData.Author.a1,
+          }),
         )
 
         api.controller.handleRequest = mockHandleRequest
 
-        const items = await endpoint.fetch()
+        let item: ApiEntityResult<FilteredResource<Author, { fields: { Author: ['name'] } }>, any>
+        try
+        {
+          item = await endpoint.getRelationshipEntity('123', 'author', {
+            fields: {Author: ['name']},
+          })
+        } catch (errors)
+        {
+          console.log(errors)
+          throw errors
+        }
+        if (item)
+        {
+          expect(item.data).toEqual({
+            id: 'a1',
+            type: 'Author',
+            name: 'Narie',
+          })
+          expect(item.data).toBeInstanceOf(Author)
+          expect(mockHandleRequest.mock.calls[0][0].href).toEqual('https://www.example.com/api/posts/123/author?fields[Author]=name');
+
+        }
+        // endpoint.fetchRelationship('123', 'comments', {}, { fields: { Comment: ['title'] } })
+      })
+    })
+
+    describe('getRelationshipCollection', () => {
```

Comment:
The name might suggest any kind of relationship might be fetched as a collection, I'd opt for `getToManyReleationship` (and `getToOneRelationship`) to align the name with the relationship field decorators (@toOne/ToManyRelationship).